### PR TITLE
ODE Compartmental JAX optimization

### PR DESCRIPTION
**Pull request by menriquez-IDM**
This change adds an optional JAX implementation of TBsim's LSHTM compartmental tuberculosis model. The goal is to accelerate workflows that repeatedly evaluate the compartmental model during calibration, history matching, and burn-in generation.

The existing SciPy implementation remains the default and continues to work exactly as before. The JAX backend is an additional execution path that provides:

- Parallel evaluation of hundreds or thousands of parameter sets
- Gradient-based calibration of compartmental model parameters
- A drop-in replacement for compartmental burn-in generation
- No changes to existing SciPy-based workflows

## Context

Several TBsim workflows repeatedly execute the LSHTM compartmental ODE model, including:

- `run_history_matching_reidentify.py`
- `run_calib_param_sweep.py`
- Burn-in generation and parameter exploration workflows

These workflows currently evaluate parameter sets sequentially by repeatedly calling `scipy.integrate.odeint` inside Python loops. While a single ODE solve is relatively inexpensive, the cumulative cost becomes significant when running large Latin Hypercube Sampling (LHS) sweeps, history-matching waves, or calibration studies involving hundreds or thousands of parameter combinations.

Although discussions with Tess suggest that the compartmental model is not currently a major part of the active TBsim workflow, the implementation is self-contained, low-maintenance, and preserves a useful capability for future calibration, burn-in, and model-fitting efforts.

## Solution

A new optional module:

```python
tbsim/compartmental/jax_ode.py
```

provides a JAX implementation of the existing LSHTM compartmental model.

```python
from tbsim.compartmental import TB_JAX_ODE, batch_run, calibrate
```

| API | Purpose |
|------|---------|
| `TB_JAX_ODE(...).run()` | Single compartmental model simulation |
| `batch_run(pars_list, ...)` | Parallel execution of many parameter sets |
| `calibrate(...)` | Gradient-based parameter fitting |

Implementation details:

- ODE equations are a direct 1:1 port of `TB_ODE`
- Double precision enabled (`jax_enable_x64`)
- JIT compilation via `jax.jit`
- Parallel parameter sweeps via `jax.vmap`
- Automatic differentiation via `jax.grad`
- Optional dependency installed through `pip install -e ".[jax]"`

The import path is guarded, so users without JAX installed see no behavioral change.

## Measured impact

The primary benefit is accelerating parameter sweeps and calibration workflows.

| Workflow | SciPy | JAX | Speedup |
|----------|--------|--------|---------|
| Single ODE solve | 15.7 ms | 0.5 ms | 30× |
| 16-parameter sweep | 0.45 s | 0.29 s | 1.6× |
| 64-parameter sweep | 1.76 s | 0.35 s | 5.0× |
| 256-parameter sweep | 7.11 s | 0.34 s | 20.8× |

### History matching

History-matching workflows evaluate large Latin Hypercube Sampling (LHS) batches of compartmental parameters.

A representative 1000-point sweep decreases from approximately:

```text
~28 seconds → ~1.4 seconds
```

This reduction compounds across successive history-matching waves and parameter-space exploration runs.

### Calibration

The new `calibrate()` API enables gradient-based optimization directly through the ODE solver.

Previously, calibration relied on derivative-free search methods such as Optuna. With JAX, gradients can be computed through the numerical integration process, enabling faster convergence for identifiable compartmental parameters and opening the door to methods such as:

- Gradient descent
- Variational inference
- Simulation-based inference workflows

### Burn-in generation

The JAX implementation preserves compatibility with the existing burn-in-to-simulation workflow.

```python
TB_JAX_ODE.handoff_state(year)
```

returns outputs compatible with:

```python
utils.ode_state_to_init_params(...)
```

allowing compartmental burn-in states to be passed directly into simulation initialization without modifications to existing code.

## Validation

The JAX implementation was validated against the existing SciPy implementation:

- Compartment trajectories agree to approximately `1e-8` relative error
- Batched (`vmap`) execution matches serial execution exactly
- Automatic gradients agree with finite-difference estimates to approximately `1e-6` relative error
- All validation tests pass

## Files changed

- `tbsim/compartmental/jax_ode.py`
- `tbsim/compartmental/__init__.py`
- `pyproject.toml`
- `tests/test_jax_accel.py`
- `tests/benchmark_jax.py`

### CODE REVIEW

**Comment by cliffckerr** [comment]
Closing, although we will be implementing compartmental Jax modeling in future!
